src/checkers/inference/DefaultSlotManager.java

```diff
@@ -94,6 +113,22 @@
     private final Map<Pair<Slot, Slot>, Integer> lubSlotPairCache;
     private final Map<Pair<Slot, Slot>, Integer> glbSlotPairCache;
 
+    /**
+     * The hashcode of the latest compilation unit. This is currently used to
+     * determine whether we should reset the defaultAnnotationsCache.
+     */
+    // TODO: we may use this to clear other caches
+    private Integer currentRootHash;
+
+    /**
+     * A map of tree hashcode to {@link AnnotationMirror} for caching
+     * a set of pre-computed default types for the current compilation unit.
+     *
+     * The reason we're storing the hashcode is to avoid holding memory
```

Comment:
Is this really a concern here? The cache is cleared after each compilation unit, so the trees aren't retained any longer than necessary.

Reply:
add setRoot to slot manager